core/identity-management/iin-agent/src/protocols/externalOperations.ts

```diff
@@ -5,26 +5,224 @@
  */
 
 import iin_agent_pb from '@hyperledger-labs/weaver-protos-js/identity/agent_pb';
-import { handlePromise} from '../common/utils';
+import * as utils from '../common/utils';
+import { LedgerBase } from '../common/ledgerBase';
+import { v4 as uuidv4 } from "uuid";
 
+// key: get using function "getSecurityDomainMapKey"
+export let securityDomainMap = new Map<string, string | iin_agent_pb.AttestedMembership>();
+export let counterAttestationsMap = new Map<string, string | iin_agent_pb.CounterAttestedMembership>();
+
+// key: Nonce
+export let foreignAgentResponseCount = new Map<string, { current: number, total: number }>();
+export let localAgentResponseCount = new Map<string, { current: number, total: number }>();
+
+export function getSecurityDomainMapKey(securityDomain: string, iinAgent: string, nonce?: string | undefined) {
+    if (nonce) {
+        return 'SEC_DOM_MAP:' + securityDomain + ':' + iinAgent + ':' + nonce;
+    } else {
+        return 'SEC_DOM_MAP:' + securityDomain + ':' + iinAgent;
+    }
+}
 
 // Handles communication with foreign IIN agents
-export const syncExternalStateFromIINAgent = async (securityDomainUnit: iin_agent_pb.SecurityDomainMemberIdentity) => {
-    console.log('syncExternalStateFromIINAgent:', securityDomainUnit.getSecurityDomain(), '-', securityDomainUnit.getMemberId());
+export const syncExternalStateFromIINAgent = async (remoteSecurityDomainUnit: iin_agent_pb.SecurityDomainMemberIdentity, securityDomain: string, memberId: string) => {
+    const remoteSecurityDomain = remoteSecurityDomainUnit.getSecurityDomain();
+    const nonce = uuidv4();
+    console.log('syncExternalStateFromIINAgent:', remoteSecurityDomain, '-', remoteSecurityDomainUnit.getMemberId(), ' nonce:', nonce);
+    
+    // const sleepTime = 2000 // 2 seconds
+    // const timeoutTime = 300 * 1000 // 5 minutes
+    // const timeoutIterations = timeoutTime/sleepTime // 5 minutes
+    
+    const requestingSecurityDomain = new iin_agent_pb.SecurityDomainMemberIdentity();
+    requestingSecurityDomain.setSecurityDomain(securityDomain);
+    requestingSecurityDomain.setMemberId(memberId);
+    
+    const request = new iin_agent_pb.SecurityDomainMemberIdentityRequest();
+    request.setRequestingNetwork(requestingSecurityDomain);
+    request.setNonce(nonce);
+    
+    // Request Attested Membership from all members of remote security_domain
+    const remoteSecurityDomainDNS = utils.getSecurityDomainDNS(remoteSecurityDomain);
+    let c = 0;
```

Comment:
Call this `count` or something descriptive, as this variable is not just a loop variable.